frontend/app/src/pages/balances/exchange/index.vue

```diff
@@ -78,11 +80,30 @@ const navigate = () => {
   vueRouter.push('/settings/api-keys/exchanges?add=true');
 };
 
+const exchangeSavingsExist: Ref<boolean> = ref(false);
+
 const exchangeDetailTabs: Ref<number> = ref(0);
+
 watch(exchange, () => {
   set(exchangeDetailTabs, 0);
+  checkSavingsData();
 });
 
+const checkSavingsData = async () => {
+  const exchangeVal = get(exchange);
+  if (exchangeVal && get(isBinance)) {
```

Comment:
IS this to trick the compiler? at least the way it works isBinance means that exchangeVal is set so the check is a bit redundant.

Can you extract maybe the core of the `isBinance` computed to a type guard like function and use that here instead of the computed?

Reply:
Updated